internal/xcuitest/config_test.go

```diff
@@ -449,3 +451,156 @@ func TestXCUITest_SortByHistory(t *testing.T) {
+		expErr        bool
+	}{
+		{
+			name: "should keep original test options when sharding is disabled",
+			project: Project{
+				Suites: []Suite{
+					{
+						Name: "no shard",
+						TestOptions: TestOptions{
+							Class: []string{"no update"},
+						},
+					},
+				},
+			},
+			expSuites: []Suite{
+				{
+					Name: "no shard",
+					TestOptions: TestOptions{
+						Class: []string{"no update"},
+					},
+				},
+			},
+		},
+		{
+			name: "should shard tests by ccy when sharding is enabled",
+			project: Project{
+				Sauce: config.SauceConfig{
+					Concurrency: 2,
+				},
+				Suites: []Suite{
+					{
+						Name:  "sharding test",
+						Shard: "concurrency",
+					},
+				},
+			},
+			content:       "test1\ntest2\n",
+			configEnabled: true,
+			expSuites: []Suite{
+				{
+					Name: "sharding test - 1/2",
+					TestOptions: TestOptions{
+						Class: []string{"test1"},
+					},
+				},
+				{
+					Name: "sharding test - 2/2",
+					TestOptions: TestOptions{
+						Class: []string{"test2"},
+					},
+				},
+			},
+		},
+		{
+			name: "should keep original test options when sharding w/o testClassesFile",
+			project: Project{
+				Sauce: config.SauceConfig{
+					Concurrency: 2,
+				},
+				Suites: []Suite{
+					{
+						Name:  "sharding test",
+						Shard: "concurrency",
+						TestOptions: TestOptions{
+							Class: []string{"test1"},
+						},
+					},
+				},
+			},
+			configEnabled: false,
+			expSuites: []Suite{
+				{
+					Name: "sharding test",
+					TestOptions: TestOptions{
+						Class: []string{"test1"},
+					},
+				},
+			},
+			expErr: true,
+		},
+		{
+			name: "should keep original test options when sharding w/ empty testClassesFile",
+			project: Project{
+				Sauce: config.SauceConfig{
+					Concurrency: 2,
+				},
+				Suites: []Suite{
+					{
+						Name:  "sharding test",
+						Shard: "concurrency",
+						TestOptions: TestOptions{
+							Class: []string{"test1"},
+						},
+					},
+				},
+			},
+			configEnabled: true,
+			content:       "",
+			expSuites: []Suite{
+				{
+					Name: "sharding test",
+					TestOptions: TestOptions{
+						Class: []string{"test1"},
+					},
+				},
+			},
+			expErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var testClassesFile string
+			if tc.configEnabled {
+				testClassesFile = createTestClassesFile(t, tc.content)
+				tc.project.Suites[0].TestClassesFile = testClassesFile
+			}
+			err := ShardSuites(&tc.project)
+			if err != nil {
+				assert.True(t, tc.expErr)
+			}
+			for i, s := range tc.project.Suites {
+				assert.True(t, cmp.Equal(s.TestOptions, tc.expSuites[i].TestOptions))
+				assert.True(t, cmp.Equal(s.Name, tc.expSuites[i].Name))
+			}
+
+			t.Cleanup(func() {
+				if testClassesFile != "" {
+					os.RemoveAll(testClassesFile)
+				}
+			})
+		})
+	}
+}
+
+func createTestClassesFile(t *testing.T, content string) string {
+	t.Helper()
+	tmpDir, _ := os.MkdirTemp("", "shard")
```

Comment:
Better to call `t.TempDir()`, as it'll get automatically cleaned up afterwards. Then you won't need the cleanup hook call you added above.